query-engine/connectors/sql-query-connector/src/cursor_condition.rs

```diff
@@ -152,17 +212,19 @@ pub fn build(query_arguments: &QueryArguments, model: &ModelRef) -> (Option<Tabl
 }
 
 // A negative `take` value signifies that values should be taken before the cursor,
-// requiring the correct comarison operator to be used to fit the reversed order.
+// requiring the correct comparison operator to be used to fit the reversed order.
 fn map_orderby_condition(
-    field: &ScalarFieldRef,
-    order: &SortOrder,
+    order_definition: &CursorOrderDefinition,
     reverse: bool,
     include_eq: bool,
 ) -> Expression<'static> {
-    let order_column = field.as_column();
-    let cmp_column = Column::from((ORDER_TABLE_ALIAS, field.db_name().to_owned()));
+    let (field, field_alias) = &order_definition.field_aliased;
+    let cmp_column = Column::from((ORDER_TABLE_ALIAS, field_alias.to_owned()));
+    let cloned_cmp_column = cmp_column.clone();
+    let order_column = order_definition.order_column.clone();
+    let cloned_order_column = order_column.clone();
```

Comment:
Couldn't figure out a better way to handle those clones in the end

Reply:
That's fine for now.